dlib/cuda/cuda_dlib.cu

```diff
@@ -1351,6 +1351,28 @@ namespace dlib
                 param.device(), params_grad.device());
         }
 
+    // ----------------------------------------------------------------------------------------
+
+        __global__ void _cuda_mish(const float* s, float* d, size_t n, const float* pp)
+        {
+            const float p = *pp;
+            for (auto i : grid_stride_range(0, n))
+            {
+                auto delta = 2*std::exp(s[i]) + std::exp(2*s[i]) + 2;
```

Comment:
This is more efficiently represented as:
```
    auto e = std::exp(s[i]);
    auto delta = 2*e + e*e + 2;
```

Reply:
Fixed.